### DISCUSSION TITLE
What should BoilR focus on next?

### DISCUSSION THREAD

**Discussion by PhilipK** [PhilipK/BoilR#80] [Polls]
I just wanted to get a feeling for what people think is the most important thing to focus on next for BoilR?

**Comment by IBNobody** [comment]
For me, Flatpak is the most important new feature for the deck. That would allow us to auto update BoilR.

**Reply by PhilipK** [reply]
I have been thinking of making a self-update solution that is cross platform, would give you what you want from the Flatpak build? 
I am still planning on doing a flatpak aswell

**Reply by IBNobody** [reply]
To give you some clarity... You had "SteamDeck" specific features on the poll. One of the biggest SD features would be providing a package on FlatHub because that would make installing, maintaining the configs for BoilR, and updating easier. Plus, it would give us an icon to launch rather than the default file icon used by the SD.

**Reply by PhilipK** [reply]
@IBNobody boilr is on flathub now, I can only test it on my laptop, but it should work on SD if flathub lives up to its name of "installs the same everywhere :-) "

**Comment by denilsonsa** [comment]
Hi! I haven't tested BoilR yet, but by looking at the README my wish is not supported yet.

It would be great if BoilR could add games installed on the system ([mostly through flathub](https://flathub.org/apps/category/Game), but potentially any games that show up at the "Games" category at the start menu). This should include grabbing images from SteamGridDb or at least using the already-installed launcher icons.

Heck, no need to restrict to games, adding stuff such as VLC or a browser or a calculator to Steam is also useful. Currently we have to switch to desktop mode, manually add them through a very basic Steam interface, and we end up of a bunch of text-only rectangles inside steam (no icons, no images whatsoever).

**Reply by PhilipK** [reply]
So a "steam deck friendly" interface for adding any app that is installed on the deck?
Any suggestion on what should be shown and what should not? Maybe just a list of everything that can be found and then make it a searchable field?

**Reply by denilsonsa** [reply]
A list of everything is a good start. Filtering and/or grouping by categories would help. AFAIK, the `.desktop` file format already includes categories, and that's how desktop environments build their start menus. You still need to research what is the format of those categories.

    fgrep -i categor ~/.local/share/applications/* /usr/share/applications/*

Grouping/filtering by shortcut location might be helpful (system-wide apps are in a different directory than user-installed apps). In fact, I haven't investigated where Steam Deck app shortcut files are located; the two directories above are just "standard" enough on most distros.

I don't know if it's possible to figure out the source of each app (e.g. package manager or flathub or something). That would be a nice filter as well, but only if it's easy enough to implement.

**Reply by denilsonsa** [reply]
As for source of images, I can think of:

* Locally-installed icons (e.g. `/usr/share/icons/` and `~/.local/share/icons/`)
* SteamGridDb
* Flathub.org (or whatever repositories the `Discover` tool might be configured to use)
* …? Anywhere else? Wikimedia commons? Mobygames? Another game/app database somewhere?

Automatically matching the app with the image may be trivial (`Icon=` in `.desktop` files), or may require some fuzzy matching (unsure, I'm just guessing here).

**Reply by PhilipK** [reply]
Well, I added flatpak as an import source , should help out with some of the apps.

**Reply by denilsonsa** [reply]
Thank you! I tried it (version 1.3.14), but it doesn't seem to list anything from flatpak/flathub, even though "Import from Flatpak" is enabled in Settings. Not sure if it's missing some permission to some folder or something. I hope you receive your Steam Deck soon enough, so that you can see stuff on the real system.

**Reply by PhilipK** [reply]
No SD here yet, hopefully next week, deckbot says I am 99.2% there.
I will have to play around with permissions and see why it is not listing flatpaks.

**Comment by NightHammer1000** [comment]
I am currently working on getting Games from Playnite, including thier Artwork, to import with BoilR.
Progress is a bit slow because I just started out with Rust, but you can expect a Pull Request sometime soon.

Playnite is my prefered Game Organizer. Especially the Artwork Scrubber is amazing.

**Reply by PhilipK** [reply]
Very cool.
Let me know if I can help, even if it is just some "pointers on rust".
If the borrow checker gives you problems just use some extra clones here and there.
This does not have to be blazingly fast, we can always optimize it later if it ever is needed.

**Comment by MiniEngineer2003** [comment]
Xbox support?

**Comment by Shimo** [comment]
Bit of a necro post. But at some point I'd love to see the option to replace the stock steam game images with animated ones where available.

**Comment by vazquez595654** [comment]
Hello,

How about support for games on a hard drive? I have many games that aren't from any stores. Would it be possible to create a section for games on a hard drive?

**Comment by jkingsbury1000** [comment]
Is EA launcher integration possible. It doesn't seem to import from EA

When I try and import from Origin it says "Failed to find Games"

**Comment by PhilipK** [comment]
I think EA should be possible, I just have not found out how yet. They seem
to have changed how they store information about which games are installed.

**Comment by NightHammer1000** [comment]
Maybe you can take a look at https://github.com/JosefNemec/PlayniteExtensions/tree/master/source/Libraries/OriginLibrary to figure out how Playnite does its detecting and build something of that.
It works on both Origin and EA App.

**Comment by PhilipK** [comment]
Thanks.
I will have a look at it when I get time.
Most likely it's going to be windows only to start, since playnite is
windows only.

**Comment by earstreet** [comment]
What do you mean with Gamepad configuration automation? 
I would prefer an option in the settings to replace the name of the imported game by the AppID. That would help me to get access to the Controller Layouts from Steam. This workaround is mentioned in this [MUO article](https://www.makeuseof.com/how-to-enable-steam-controller-configuration-non-steam-games/). Unfortunately I don't know another.
Actually I have to replace it every time I import a new game with BoilR because it overwrites my changes. Would be really nice if you could make such an option available. Thanks!

**Reply by PhilipK** [reply]
If you doubleclick a game in the import menu you can rename the imported game, and it will be respected on the next import as well.

**Reply by earstreet** [reply]
Thx for the answer. I didn't know this feature. So I just give it a try.
Unfortunately the program lost all images after the change of the name to the AppID in BoilR. I think it can't find the games in SteamDB by the AppID. But when I use the number in SteamDB search window it's working. So the number must be correct. Maybe you can also take a look at it.

**Comment by quentincaffeino** [comment]
Better ui? Maybe even use tauri. This might allow people with little knowledge of rust to contribute to program ui.
